test/integration/Client/client_lib_thread_interdependency_test.cpp

```diff
@@ -0,0 +1,499 @@
+                     "trying to destroy a story on a state different from STORY_ACQUIRED state.", tid);
+        }
+    }
+    else if(ret == chronolog::CL_ERR_NOT_EXIST)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_DESTROYED ||
+           static_cast<ThreadState>(shared_state[tid]) == ThreadState::CHRONICLE_DESTROYED)
+        {
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_NOT_EXIST return value when "
+                     " trying to destroy a Story on the STORY_DESTROYED or CHRONICLE_DESTROYED state.", tid);
+        }
+        else
+        {
+            LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_NOT_EXIST return value when "
+                      "trying to destroy a story on a state different than STORY_DESTROYED or CHRONICLE_DESTROYED "
+                      "state.", tid);
+        }
+    }
+    else
+    {
+        LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- the return value is an unexpected one", tid);
+    }
+}
+
+void check_chronicle_destroyed(int tid, int ret)
+{
+    std::lock_guard <std::mutex> lock(shared_state_mutex);
+    if(ret == chronolog::CL_SUCCESS)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_RELEASED ||
+           static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_DESTROYED)
+        {
+            shared_state[tid] = static_cast<int>(ThreadState::CHRONICLE_DESTROYED);
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- State updated to: {}", tid, get_state_name(
+                    ThreadState::CHRONICLE_DESTROYED));
+            update_chronicle_threads_state(tid, ThreadState::CHRONICLE_DESTROYED);
+        }
+        else
+        {
+            LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_SUCCESS return value when trying "
+                      "to destroy chronicle on a state different than STORY_RELEASED", tid);
+        }
+    }
+    else if(ret == chronolog::CL_ERR_ACQUIRED)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_ACQUIRED)
+        {
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_ACQUIRED return value when "
+                     "trying to destroy chronicle on STORY_ACQUIRED state", tid);
+        }
+        else
+        {
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_ACQUIRED return value when "
+                     "trying to destroy chronicle on a state different than STORY_ACQUIRED", tid);
+        }
+    }
+    else if(ret == chronolog::CL_ERR_NOT_EXIST)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_DESTROYED ||
+           static_cast<ThreadState>(shared_state[tid]) == ThreadState::CHRONICLE_DESTROYED)
```

Comment:
This contradicts with the line where I left my previous comment.

Reply:
yes. There is a contradiction here. The only option where destroy chronicle should report a `CL_ERR_NOT_EXIST` is if the chronicle has already been destroyed, you can destroy stories inside and still keep the chronicle. So this is a mistake.